dias/analyzers/flag_rfi_analyzer.py

```diff
@@ -0,0 +1,592 @@
+                        dset.attrs['threshold'] = self.threshold_mad
+
+                        dset = handler.create_dataset('auto', data=auto)
+                        dset.attrs['axis'] = ax
+
+                        dset = handler.create_dataset('ndev', data=ndev)
+                        dset.attrs['axis'] = ax
+
+                    # Update data index database
+                    self.update_data_index(data.time[0], data.time[-1],
+                                           filename=output_file)
+
+                    # Update prometheus metrics
+                    ndf = len(files)
+                    self.file_counter.inc(ndf)
+
+                    time_per_file = int((time.time() - self.file_start_time) /
+                                        float(ndf))
+                    self.file_timer.set(time_per_file)
+
+                    for ss, lbl in enumerate(stack):
+                        self.fraction_masked_missing.labels(stack=lbl).set(
+                                    _fraction_flagged(mask_missing[:, ss, :],
+                                                      logical_not=True))
+
+                        self.fraction_masked_before.labels(stack=lbl).set(
+                                    _fraction_flagged(mask_before[:, ss, :],
+                                                      logical_not=True))
+
+                        self.fraction_masked_after.labels(stack=lbl).set(
+                                    _fraction_flagged(mask_after[:, ss, :],
+                                                      logical_not=True))
+
+                    # Garbage collect
+                    del data
+                    gc.collect()
+
+        # Increment run counter.
+        self.run_counter.inc()
+        self.run_timer.set(int(time.time() - self.run_start_time))
+
+    def update_data_index(self, start, stop, filename=None):
+        """Add row to data index database.
+
+        Update the data index database with a row that
+        contains the name of the file and the span of time
+        the file contains.
+
+        Parameters
+        ----------
+        start : unix time
+            Earliest time contained in the file.
+        stop : unix time
+            Latest time contanied in the file.
+        filename : str
+            Name of the file.
+        """
+        # Parse arguments
+        dt_start = ephemeris.unix_to_datetime(ephemeris.ensure_unix(start))
+        dt_stop = ephemeris.unix_to_datetime(ephemeris.ensure_unix(stop))
+
+        relpath = None
+        if filename is not None:
+            relpath = os.path.relpath(filename, self.write_dir)
+
+        # Insert row for this file
+        try:
+            cursor = self.data_index.cursor()
+            cursor.execute("INSERT INTO files VALUES (?, ?, ?)",
+                           (dt_start, dt_stop, relpath))
+
+        except Exception as ex:
+            self.log.error("Could not perform database insert: %s" % ex)
```

Comment:
Do you want to raise after this? We will likely add an alert on analyzers failing. This on would just print an error but go through as a normal task execution in this case here.